src/web/client/test/integration/errorHandler.test.ts

```diff
@@ -0,0 +1,634 @@
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+import powerPlatformExtensionContext from "../../../client/WebExtensionContext";
+import { schemaKey } from "../../schema/constants";
+import { telemetryEventNames } from "../../telemetry/constants";
+
+import {
+    queryParameters,
+    PORTALS_FOLDER_NAME_DEFAULT,
+} from "../../common/constants";
+
+import {
+    showErrorDialog,
+    removeEncodingFromParameters,
+    checkMandatoryParameters,
+    checkMandatoryPathParameters,
+    checkMandatoryQueryParameters,
+} from "../../common/errorHandler";
+describe("errorHandler", () => {
+    afterEach(() => {
+        Sinon.restore();
+    });
+    it("showErrorDialog_shouldCallShowErrorMessage", () => {
+        //Act
+        const errorString = "this is error message";
+        const detailMessage = "not able to open";
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+
+        //Action
+        showErrorDialog(errorString, detailMessage);
+
+        //Assert
+        const options = { detail: detailMessage, modal: true };
+        assert.calledOnceWithExactly(
+            _mockShowErrorMessage,
+            errorString,
+            options
+        );
+    });
+
+    it("showErrorDialog_shouldCallShowErrorMessage_whenDetailMessagesIsNotPassed", () => {
+        //Act
+        const errorString = "this is error message";
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+
+        //Action
+        showErrorDialog(errorString);
+
+        //Assert
+        const options = { detail: undefined, modal: true };
+        assert.calledOnceWithExactly(
+            _mockShowErrorMessage,
+            errorString,
+            options
+        );
+    });
+
+    it("removeEncodingFromParameters_shouldDecodeSCHEMAVERSIONAndWEBSITENAME", () => {
+        //Act
+        const queryParamsMap = new Map<string, string>([
+            [schemaKey.SCHEMA_VERSION, "3A%20test_file_name"],
+            [queryParameters.WEBSITE_NAME, "3A%20test_test_webSiteName"],
+        ]);
+
+        //Action
+        removeEncodingFromParameters(queryParamsMap);
+
+        //Assert
+        const websiteName = queryParamsMap.get(
+            queryParameters.WEBSITE_NAME
+        ) as string;
+
+        const schemaFileName = queryParamsMap.get(
+            schemaKey.SCHEMA_VERSION
+        ) as string;
+
+        expect(schemaFileName).eq("3A test_file_name");
+        expect(websiteName).eq("3A test_test_webSiteName");
+    });
+
+    it("removeEncodingFromParameters_shouldMapPORTALSFOLDERNAMEDEFAULT_whenWebsiteNameIsBlank_", () => {
+        //Act
+        const queryParamsMap = new Map<string, string>([
+            [schemaKey.SCHEMA_VERSION, "3A%20test_file_name"],
+            [queryParameters.WEBSITE_NAME, ""],
+        ]);
+
+        //Action
+        removeEncodingFromParameters(queryParamsMap);
+
+        //Assert
+        const websiteName = queryParamsMap.get(
```

Comment:
create another variable with portalfoldername and then do assertions with portalfoldername. Keep websiteName as it is

Reply:
Done